### PR TITLE
Submit queue batch fixes

### PR DESCRIPTION
**Pull request by rmmh**
- Reveal BaseRef so batches can be filtered to just "master"
- Stop polling for status if the PR is closed underneath us.

### CODE REVIEW

**Comment by k8s-oncall** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/2107)

File: mungegithub/github/github.go [hunk 1]
```diff
@@ -1228,6 +1228,12 @@ func (obj *MungeObject) doWaitStatus(pending bool, requiredContexts []string, c
 			glog.V(4).Infof("PR# %d is pending, waiting for %f seconds", *obj.Issue.Number, sleepTime.Seconds())
 		}
 		time.Sleep(sleepTime)
+
+		// If it has been closed, assume that we want to break from the poll loop early.
+		obj.Refresh()
```
Comment (rmmh) [inline mungegithub/github/github.go]: ~4 API calls per minute, which is pretty tiny when the SQ regularly uses 5 API calls per second.

Comment (spxtr) [inline mungegithub/github/github.go]: I'm not sure if that's a great argument, but okay.

**Pull request merged**